lib/most_frequent_integer.rb

```diff
@@ -2,6 +2,26 @@
 # frequent integer. These integers might be positive or negative. If no most-frequent integer
 # exists, return nil.
 
-def find_most_frequent_integer(arr)
-  # put your code here
+#def find_most_frequent_integer(arr)
+#    num ={
+#  num = arr.max_by{|k,v| v}
+#  return num.first
+#end
+
+def create_collection_of_numbers(nums)
+  h={}
+  nums.each do |n|
+    if h.has_key?(n) == false
+      h[n]=1
+    elsif h.has_key?(n) == true
+      h[n]= h[n] +1
+    end
+  end
+  return h
 end
+
+def find_most_frequent_integer(nums)
+  h = create_collection_of_numbers(nums)
+  arr = h.sort_by{|k,v| v}
```

Comment:
This seems like it would be a good way to accomplish this problem, but it has one issue. Let's take, for example, this array: `[8, 9]`. What's the most frequent integer in that array? You could say it's a tie, or you might say there is not a most frequent integer at all. The latter is what the [instructions](https://github.com/omahacodeschool/ruby-toy__most-frequent-integer) of this assignment indicate. So that's one issue: Sorting by frequency will fail to detect a tie.

Here is how I would improve your algorithm:

1. Count each element's frequency and store those sets of information in a hash. (This is what your `create_collection_of_numbers` method does successfully.)
2. Figure out what the highest frequency in that hash is.
3. Figure out if more than one element in the hash has that frequency. Because it more than one does, then there's a tie and you should return `nil`.
4. If only one element in the hash has that frequency, find it and return it.

Try converting this algorithm into code, and we'll take it from there.